### PR TITLE
feat: Remove magic strings from Source type

### PR DESCRIPTION
**Pull request by pmengelbert**
Replaces PR #57 , see #57 for all comments which have been addressed by now.

There was a snafu with the Azure/dalec repo going public, so I had to re-fork.

### CODE REVIEW

File: spec.go [hunk 1]
```diff
@@ -179,71 +177,95 @@ type ImageConfig struct {
...
+// Source defines a source to be used in the build.
+// A source can be a local directory, a git repositoryt, http(s) URL, etc.
+type Source struct {
+	// This is an embedded union representing all of the possible source types.
+	// Exactly one must be non-nil, with all other cases being errors.
+	//
+	// === Begin Source Variants ===
+	DockerImage *SourceDockerImage `yaml:"image,omitempty" json:"image,omitempty"`
+	Git         *SourceGit         `yaml:"git,omitempty" json:"git,omitempty"`
+	HTTPS       *SourceHTTPS       `yaml:"https,omitempty" json:"https,omitempty"`
```
Comment (cpuguy83) [inline spec.go:256]: Nit: this should probably just be `http`.
Tried to think of a better name than that but am coming up short.

Comment (pmengelbert) [reply spec.go:256]: The field name? The Struct name? The json tag? All of the above?

Comment (cpuguy83) [reply spec.go:256]: All of the above.

Comment (pmengelbert) [reply spec.go]: done

**Review by cpuguy83 — APPROVED** [review]

**Pull request merged**